doc/release-notes.md

```diff
@@ -23,492 +26,577 @@ shut down (which might take a few minutes for older versions), then run the
+connection count for masternodes. These intra-quorum connections are less resource
+demanding than normal p2p connections as they only exchange LLMQ/masternode related
+messages, but the hardware and network requirements will still be higher than before.
+
+This change will at first only be activated for the smaller LLMQs (50 members) and
+then may later be activated for larger quorums (400 members).
+
+This is also controlled via `SPORK_21_QUORUM_ALL_CONNECTED`.
+
+Masternode Connection Probing
+-----------------------------
+While each LLMQ member must have a connection to each other member, it's not necessary
+for all members to actually connect to all other members. This is because only
+one of a pair of two masternodes need to initiate the connection while the other one can
+wait for an incoming connection. Probing is done in the case where a masternode doesn't
+really need an outbound connection, but still wants to verify that the other side
+has its port open. This is done by initiating a short lived connection, waiting
+for `MNAUTH` to succeed and then disconnecting again.
+
+After this process, each member of a LLMQ knows which members are unable to accept
+connections, after which they will vote on these members to be "bad".
+
+Masternode Minimum Protocol Version Checks
+------------------------------------------
+Members of LLMQs will now also check all other members for minimum protocol versions
+while in DKG. If a masternode determines that another LLMQ member has a protocol
+version that is too low, it will vote for the other member to be "bad".
+
+PoSe punishment/banning
+-----------------------
+If 80% of all LLMQ members voted for the same member to be bad, it is excluded
+in the final stages of the DKG. This causes the bad masternode to get PoSe punished
+and then eventually PoSe banned.
+
+Network performance improvements
+--------------------------------
+This version of Dash Core includes multiple optimizations to the network and p2p message
+handling code. The most important one is the introduction of `epoll` on linux-based
+systems. This removes most of the CPU overhead caused by the sub-optimal use of `select`,
+which could easily use up 50-80% of the CPU time spent in the network thread when many
+connections were involved.
+
+Since these optimizations are exclusive to linux, it is possible that masternodes hosted
+on windows servers will be unable to handle the network load and should consider migrating
+to a linux based operating system.
+
+Other improvements were made to the p2p message handling code, so that LLMQ
+related connections do less work than full/normal p2p connections.
+
+Wallet files
+------------
+The `--wallet=<path>` option now accepts full paths instead of requiring
+wallets to be located in the -walletdir directory.
+
+If `--wallet=<path>` is specified with a path that does not exist, it will now
+create a wallet directory at the specified location (containing a wallet.dat
```

Comment:
Probably not absolutely needed but could make it a bit more clear together with the next comment where we have `database/log.?????????` instead of database/log.?????????? which imo looks weird in the middle of a text. 
```suggestion
create a wallet directory at the specified location (containing a `wallet.dat`
```

Or maybe just wrap all of them in "" ?